tests/unit/test_client.py

```diff
@@ -686,3 +703,89 @@ def test_list_instances_w_options(self):
             retry=mock.ANY,
             timeout=mock.ANY,
         )
+
+    def test_observability_options_propagated_extended_tracing_off(self):
+        self.__test_observability_options(True)
+
+    def test_observability_options_propagated(self):
+        self.__test_observability_options(False)
+
+    def __test_observability_options(self, enable_extended_tracing):
+        if not hasOtelInstalled:
+            return
+
+        # This test needs the spanner emulator setup so as to run.
```

Comment:
unit tests are never run against emulator which means this test will not be executed at all.
Ref: https://github.com/googleapis/python-spanner/blob/b2c69d433a129a6313b83e9978a2b7bae73814cf/.github/workflows/integration-tests-against-emulator.yaml#L28

Can you move these to system tests (i.e under tests/system)? You can create a new file to avoid complexity

Reply:
You can annotate the test with below to make sure to run only on emulator

@pytest.mark.skipif(
   not  _helpers.USE_EMULATOR, reason="Emulator does not support DML Returning."
)

Reply:
Done, thanks!